### PR TITLE
[GATE 1734] Unit test coverage

### PR DESCRIPTION
**Pull request by dpitch40**
All Java client library endpoints covered, all unit tests passing (when running offline and online).

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
What are you using for tracking coverage? Can you add some reporting to it so we know the number? I briefly added Cobertura (see https://www.mkyong.com/qa/maven-cobertura-code-coverage-example/) and seems to hover around 50%

**Comment by dpitch40** [comment]
My understanding of this task was that I was simply aiming to cover every endpoint implemented in the Java client library. So I made a list of all the library methods that hit our API and checked them off as they were covered by unit tests. I will ask Nick to confirm.

**Comment by dpitch40** [comment]
Nick and I don't think 100% line coverage is worth the time investment now. His intention was simply to bring unit test coverage up to the standard of other libraries, and it now probably greatly exceeds that of the other libraries.

File: src/main/java/net/vivialconnect/model/VivialConnectResource.java [hunk 1]
```diff
@@ -146,6 +146,7 @@ protected static String formatURLForResource(String resourceName){
 
     private static VivialConnectException handleException(Exception e){
         VivialConnectException vce = null;
+        System.out.println(e.toString());
```
Comment (nphcbm) [inline src/main/java/net/vivialconnect/model/VivialConnectResource.java]: having all these print statements (even commented out ones) seems like a bad idea.

File: src/main/java/net/vivialconnect/model/format/ListFormatter.java [hunk 2]
```diff
@@ -7,14 +7,17 @@
     private StringBuilder builder;
 
 
-    public ListFormatter(){
-        this.builder = new StringBuilder();
-        this.builder.append("[");
-    }
+    // public ListFormatter(){
```
Comment (nphcbm) [inline src/main/java/net/vivialconnect/model/format/ListFormatter.java]: Why are we leaving commented out code in our codebase?

**Pull request merged**